nntrainer/src/pooling2d_layer.cpp

```diff
@@ -54,7 +54,52 @@ int Pooling2DLayer::setSize(int *size,
 
 int Pooling2DLayer::setProperty(std::vector<std::string> values) {
   int status = ML_ERROR_NONE;
-  // NYI
+
+  for (unsigned int i = 0; i < values.size(); ++i) {
+    std::string key;
+    std::string value;
+
+    status = getKeyValue(values[i], key, value);
+    NN_RETURN_STATUS();
+    unsigned int t = parseLayerProperty(key);
+    switch (static_cast<PropertyType>(t)) {
+    case PropertyType::pooling:
+      pooling_type = (PoolingType)parseType(value, TOKEN_POOLING);
+      if (pooling_type == PoolingType::unknown) {
+        ml_loge("Error: Unknown pooling type");
+        return ML_ERROR_INVALID_PARAMETER;
+      }
+      break;
+    case PropertyType::pooling_size:
+      status = getValues(POOLING2D_DIM, value, (int *)(pooling_size));
+      if (pooling_size[0] == 0 || pooling_size[1] == 0) {
+        ml_loge("Error: pooling_size must be greater than 0");
+        return ML_ERROR_INVALID_PARAMETER;
+      }
+      NN_RETURN_STATUS();
```

Comment:
IMO, `NN_RETURN_STATUS()` should be performed before looking at the parsed values.
Most likely, this won't change the results but it is better in reading.

Reply:
fixed. Thanks.